libbeat/common/kubernetes/metadata/metadata.go

```diff
@@ -50,3 +50,20 @@ func WithLabels(kind string) FieldOptions {
 		safemapstr.Put(meta, strings.ToLower(kind)+".labels", meta["labels"])
 	}
 }
+
+// GetPodMetaGen is a wrapper function that creates a metaGen for pod resource and has embeeded
+// nodeMetaGen and namespaceMetaGen
+func GetPodMetaGen(
+	cfg *common.Config,
+	podWatcher kubernetes.Watcher,
+	nodeWatcher kubernetes.Watcher,
+	namespaceWatcher kubernetes.Watcher,
+	metaConf *AddResourceMetadataConfig,
+	hostname string) MetaGen {
```

Comment:
We are passing the `hostname` in many places, I wonder if we may be using the wrong hostname when using cluster scope.

Reply:
++ good catch!